modules/administration-guide/pages/repairing-the-devworkspace-operator-on-openshift.adoc

```diff
@@ -0,0 +1,80 @@
+
+[id="repairing-the-devworkspace-operator-on-openshift_{context}"]
+= Repairing the {devworkspace} Operator on OpenShift
+
+Under certain conditions, such as https://docs.openshift.com/container-platform/4.10/operators/understanding/olm/olm-understanding-olm.html[OLM] restart or cluster upgrade, the {prod2} Operator for {prod-short} might automatically install the {devworkspace} Operator even when it is already present on the cluster. In that case, you can repair the {devworkspace} Operator on OpenShift as follows:
+
+.Prerequisites
+
+* An active `{orch-cli}` session as a cluster administrator to the destination OpenShift cluster. See {orch-cli-link}.
+
+* On the *Installed Operators* page of the {platforms-name} web console, you see multiple entries for the {devworkspace} Operator or one entry that is stuck in a loop of *Installing* and *InstallSucceeded*.
+
+.Procedure
+
+. Delete the `devworkspace-controller` namespace that contains the failing pod.
+
+. Update `DevWorkspace` and `DevWorkspaceTemplate` Custom Resource Definitions (CRD) by setting the conversion strategy to `None` and removing the next several lines of the removed strategy:
++
+[source,yaml]
+----
+spec:
+  ...
+  conversion:
+    strategy: None
+status:
+...
+----
++
+TIP: You can find and edit the `DevWorkspace` and `DevWorkspaceTemplate` CRDs in the *Administrator* perspective of the OpenShift web console by searching for `DevWorkspace` in menu:Administration[CustomResourceDefinitions].
++
+NOTE: The `DevWorkspaceOperatorConfig` and `DevWorkspaceRouting` CRDs  have the conversion strategy set to `None` by default.
+
+. Remove the {devworkspace} Operator subscription:
++
+----
+$ oc delete sub devworkspace-operator
+----
+
+. Get the {devworkspace} Operator CSVs in the __<devworkspace-operator.vX.Y.Z>__ format:
++
+----
+$ oc get csv | grep devworkspace
+----
+
+. Remove each {devworkspace} Operator CSV:
++
+[subs="+quotes"]
+----
+$ oc delete csv __<devworkspace-operator.vX.Y.Z>__ <1>
+----
+<1> For example, `devworkspace-operator.v0.15.1` or `devworkspace-operator.v0.15.2`.
+
+. Re-create the {devworkspace} Operator subscription:
++
+[subs="+attributes"]
+----
+$ cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: devworkspace-operator
+  namespace: openshift-operators
+spec:
+  channel: fast
+  name: devworkspace-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  installPlanApproval: Manual
+  startingCSV: devworkspace-operator.v{devworkspace-operator-version-patch}
+EOF
+----
+
+. In the *Administrator* perspective of the OpenShift web console, go to menu:Operators[Installed Operators] and select the following for the *{devworkspace} Operator*: menu:Upgrade available[Preview InstallPlan > Approve].
+
+. In the *Administrator* perspective of the OpenShift web console, go to menu:Operators[Installed Operators] and verify the *Succeeded* status of the *{devworkspace} Operator*.
```

Comment:
I have end up with "Pending" subscription status when fix an upgrade from DWO 0.15.2 to 0.15.3:
> Failed
> error validating existing CRs against new CRD's schema for "devworkspacetemplates.workspace.devfile.io": error listing resources in GroupVersionResource schema.GroupVersionResource{Group:"workspace.devfile.io", Version:"v1alpha1", Resource:"devworkspacetemplates"}: conversion webhook for workspace.devfile.io/v1alpha2, Kind=DevWorkspaceTemplate failed: Post "https://devworkspace-controller-manager-service.openshift-operators.svc:443/convert?timeout=30s": no endpoints available for service "devworkspace-controller-manager-service"
![Screenshot from 2022-08-17 02-28-25](https://user-images.githubusercontent.com/1197777/185002452-4a997556-6d77-4185-a71d-a32b8357e0f8.png)

Reply:
probably you have not patched CRDs with the `None` strategy - https://github.com/eclipse-che/che-docs/pull/2419/files#diff-5b6dbd77da6b2681108fbf5d3a311fc960e537018e961aab4b2da7c07527bec3R18-R36

Reply:
Patching the CRD is step 2 of this procedure. Did it not work?

Reply:
@ibuziuk : you are right - I have fixed DevWorkspace and DevWorkspaceTemplate objects themselves, not their CRDs.

Reply:
@ibuziuk, @max-cx : DevWorkspace Operator had been installed successfully after the proper patch of CRDs: https://github.com/eclipse-che/che-docs/pull/2419/files#r947608670

![Screenshot from 2022-08-17 12-25-02](https://user-images.githubusercontent.com/1197777/185084386-e2400518-c09a-4fe0-8bd0-93ceb91acf1e.png)

DS 3.0.0 was able to start workspace.

Reply:
Great! I'm resolving this thread then.